app/src/main/java/com/mstudio/superheroarch/usecase/GetAllCharactersUseCase.kt

```diff
@@ -0,0 +1,18 @@
+package com.mstudio.superheroarch.usecase
+
+import com.mstudio.superheroarch.presentation.model.CharacterData
+import com.mstudio.superheroarch.remotedatasource.model.toCharacterData
+import com.mstudio.superheroarch.repository.RickAndMortyRepository
+
+class GetAllCharactersUseCase(
+    private val repository: RickAndMortyRepository
+) {
+
+    suspend fun getAllCharacters(): List<CharacterData>? {
+        try {
+            return repository.getCharacters()?.map { it.toCharacterData() }
```

Comment:
If repository would have a non nullable return type, you could simplify this. In general, error management could be improved. Nothing too important now, but try to have it in mind always.